lib/transport/selenium-webdriver/browserstack.js

```diff
@@ -9,19 +10,19 @@ class Browserstack extends BaseDriver {
 
 
   get accessKey() {
-    return this.bstackOptions.accessKey;
+    return this.settings.desiredCapabilities['bstack:options'].accessKey;
   }
 
   get username() {
-    return this.bstackOptions.userName;
+    return this.settings.desiredCapabilities['bstack:options'].userName;
   }
 
   get build() {
-    return this.bstackOptions.buildName;
+    return this.settings.desiredCapabilities['bstack:options'].buildName;
```

Comment:
```suggestion
    return this.settings.desiredCapabilities.buildName;
```

Reply:
URL still contains undefined if buildname is present in desiredCapabilities but not present in 'bstack:options'.